python/semantic_kernel/agents/history/chat_history_summarization_reducer.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from typing import Any
+
+from pydantic import Field
+
+from semantic_kernel.agents.history.chat_history_reducer import ChatHistoryReducer
+from semantic_kernel.agents.history.chat_history_reducer_extensions import (
+    SUMMARY_METADATA_KEY,
+    extract_range,
+    locate_safe_reduction_index,
+    locate_summarization_boundary,
+)
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.const import DEFAULT_SERVICE_NAME
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.exceptions.agent_exceptions import AgentChatHistoryReducerException
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SUMMARIZATION_INSTRUCTIONS = """
+        Provide a concise and complete summarization of the entire dialog that does not exceed 5 sentences
+
+        This summary must always:
+        - Consider both user and assistant interactions
+        - Maintain continuity for the purpose of further dialog
+        - Include details from any existing summary
+        - Focus on the most significant aspects of the dialog
+
+        This summary must never:
+        - Critique, correct, interpret, presume, or assume
+        - Identify faults, mistakes, misunderstanding, or correctness
+        - Analyze what has not occurred
+        - Exclude details from any existing summary
+        """
+
+
+class ChatHistorySummarizationReducer(KernelBaseModel, ChatHistoryReducer):
```

Comment:
Why don't we inherit from ChatHistory itself? Then its instantly usable in all scenarios and agents check for a reduce method and use that when supplied and users in their code can just use it and call reduce themselves, it should then move out of the agents folder, but I think that makes much more sense!

Reply:
That makes sense. I worked on moving to that architecture today. I'll push the changes soon. In the process of changing things around, I encountered some nasty circular dependencies that seem to have existed in our code for a while. Part of it was related to the `FunctionCallBehavior`. I've decided, that having that marked as `deprecated` for more than 6 months now, I ripped it out. I'll make a note in the PR description above.

In any case, I've decided to make the `ChatHistoryReducer` a subclass of `ChatHistory`, as opposed to each class implementing `ChatHistory` on their own. This makes sure that all reducers can be treated as interchangeable with `ChatHistory`. It also reduces redundant inheritance declarations in individual reducers, so we aren't "repeating ourselves."